scripts/mixins/job_special.lua

```diff
@@ -0,0 +1,52 @@
+-- for mobs with job-specific special ability
+
+require("scripts/globals/mixins");
+require("scripts/globals/status");
+
+g_mixins = g_mixins or {};
+
+g_mixins.job_special = function(mob)
+
+    mob:addListener("SPAWN", "JOB_SPECIAL_SPAWN", function(mob)
+        local defaultAbility = {
+            [JOBS.WAR] = jobSpec.MIGHTY_STRIKES,
+            [JOBS.MNK] = jobSpec.HUNDRED_FISTS,
+            [JOBS.WHM] = jobSpec.BENEDICTION,
+            [JOBS.BLM] = jobSpec.MANAFONT,
+            [JOBS.RDM] = jobSpec.CHAINSPELL,
+            [JOBS.THF] = jobSpec.PERFECT_DODGE,
+            [JOBS.PLD] = jobSpec.INVINCIBLE,
+            [JOBS.DRK] = jobSpec.BLOOD_WEAPON,
+            [JOBS.BST] = jobSpec.FAMILIAR,
+            [JOBS.BRD] = jobSpec.SOUL_VOICE,
+            [JOBS.SAM] = jobSpec.MEIKYO_SHISUI,
+            [JOBS.NIN] = jobSpec.MIJIN_GAKURE,
+            [JOBS.DRG] = jobSpec.CALL_WYVERN,
+            [JOBS.SMN] = jobSpec.ASTRAL_FLOW,
+        };
+
+        mob:setLocalVar("usedJobSpec", 0);
+        mob:setLocalVar("useJobSpecAtPct", math.random(40,60));
+        local ability = defaultAbility[mob:getMainJob()];
```

Comment:
We really do not want to assume a mobs mainjob will decide what 2hr is uses. There are ,many examples of that not being the case.

Reply:
I'd rather we just specify which 2hr we are trying to call as an argument to a function.

Reply:
I think it's fine to have an overrideable default be their main job (as I was saying in my other comment 2 lines down)

Reply:
only saw you mention subjob sorta skimmed over most of your comment to lines down because the bulk of it isn't talking about this. something set the percent something localvar something override something about when the listener runs.

I missed one out of 4, spoke about the only bit that worries me. I disagree about bothering with a default instead of being explicit about what we want to happen but you da boss.

Reply:
it'll use the default main job ability if one isn't specific in onMobSpawn().  figure that lets us write less code for the most common case, but gives flexibility for specific mobs that use weird 1hrs